components/layout_2020/display_list/stacking_context.rs

```diff
@@ -896,7 +896,10 @@ struct ReferenceFrameData {
 
 impl BoxFragment {
     fn get_stacking_context_type(&self) -> Option<StackingContextType> {
-        if self.style.establishes_stacking_context() {
+        if self
+            .style
+            .establishes_stacking_context(self.base.flags.contains(FragmentFlags::IS_REPLACED))
```

Comment:
Could pass the `self.base.flags` and check `contains(FragmentFlags::IS_REPLACED)` there. Bool parameters aren't great anyways.